contracts/contracts/VerificationGateway.sol

```diff
@@ -112,6 +113,33 @@ contract VerificationGateway
         require(verified, "VG: Sig not verified");
     }
 
+    function validateSignature(
+        bytes32 hash,
+        bytes memory signature
+    ) public view returns (bool verified) {
+        IWallet wallet = IWallet(msg.sender);
+        bytes32 existingHash = hashFromWallet[wallet];
+        require(
+            (IWallet(msg.sender) == walletFromHash[existingHash]),
+            "VG: not called from wallet"
+        );
+
+        uint256[BLS_KEY_LEN] memory publicKey = BLSPublicKeyFromHash[existingHash];
+
+        bytes memory concatenatedHash = bytes.concat(hash);
+
+        uint256[2] memory message = blsLib.hashToPoint(
+            BLS_DOMAIN,
+            concatenatedHash
+        );
+
+        require(signature.length == 64, "VG: Sig bytes length must be 64");
+        uint256[2] memory decodedSignature = abi.decode(signature, (uint256[2]));
+
+        verified = blsLib.verifySingle(decodedSignature, publicKey, message);
+        require(verified, "VG: Sig not verified");
```

Comment:
The 1271 reference implementation returns `0xffffffff`, which I guess could be any other value. Is throwing preferred?

Reply:
Yeah noticed that. My original thought was to use a similar require statement to the verify function in the same file. So I original intent was consistency. 

From different implementations:
[0xsequence - returns only happy path magic value (0x1626ba7e), but doesn't throw if signature is incorrect](https://github.com/0xsequence/wallet-contracts/blob/50204fd69535207cfbf5894c9ce4c100a3f82b91/src/contracts/modules/commons/ModuleAuth.sol#L147)
[ambire - uses 0xffffffff](https://github.com/AmbireTech/wallet/blob/f6c952c8b7ad0c0a5d48d7ba22ab265d8a5429d0/contracts/QuickAccManager.sol#L124)
[etherspot - uses 0xffffffff](https://github.com/etherspot/etherspot-contracts/blob/3d07a849a00832d4cad7ac6d56c9ab9d31b2e772/src/common/account/AccountImplementationV1.sol#L20)
[biconomy - uses 0xffffffff](https://github.com/bcnmy/scw-contracts/blob/ac19b1569dca05f76eec233f7b686776141f231f/contracts/smart-contract-wallet/SmartAccount.sol#L798)

So maybe throwing is not preferred and returning 0xffffffff would be more consistent with other implementations?

Reply:
changed to `0xffffffff`, to match erc reference implementation, and implementations from other projects